app/api/v1/coffee_shops_route.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class CoffeeShopsRoute < BaseRoute
+      route do |r|
+        r.get do
+          csv_url = ENV.fetch('COFFEE_SHOPS_CSV_URL', 'https://raw.githubusercontent.com/Agilefreaks/test_oop/master/coffee_shops.csv')
+          user_coordinates = permit_params(r.params, %w[x y]).transform_keys(&:to_sym)
+          csv_data = Services::FetchCsvService.new(csv_url).call
+          coffee_shops = Services::ParseCsvService.new(csv_data).call
```

Comment:
is there a benefit of separating fetching and parsing operations?

Reply:
SRP, and it's more flexible like this because if the way you fetch or parse data changes, you only need to update the respective service without affecting the other.

Reply:
The client has too many responsibilities now. How this will age when we'd need to add caching, circuit breaking, throttling, storing, etc.?

Reply:
Related with the not so good choice of Command Pattern...
> Future concerns like caching, throttling, or circuit breaking can be added inside the CoffeeShopWorkflowService without affecting the client or individual services.